src/OpenTelemetry/Logs/OpenTelemetryLoggerProvider.cs

```diff
@@ -47,6 +49,8 @@ internal OpenTelemetryLoggerProvider(OpenTelemetryLoggerOptions options)
         {
             this.Options = options ?? throw new ArgumentNullException(nameof(options));
 
+            this.Resource = options.ResourceBuilder.Build();
```

Comment:
Could we make OpenTelemetryLoggerProvider implement `BaseProvider`, (similar to `TracerProviderSdk`) and then modify `ProviderExtensions.GetResource` to return the resource?

Reply:
This can avoid storing Resource inside LogRecord, but still make it available for Processor/Exporter.

Reply:
I've updated the PR, please take another look.